x-pack/plugins/osquery/public/agents/agents_table.tsx

```diff
@@ -5,158 +5,225 @@
+import {
+  getNumAgentsInGrouping,
+  generateAgentCheck,
+  getNumOverlapped,
+  generateColorPicker,
+} from './helpers';
+
+import { AGENT_GROUP_KEY, SelectedGroups, AgentOptionValue, GroupOptionValue } from './types';
+
+export interface AgentsSelection {
+  agents: string[];
+  allAgentsSelected: boolean;
+  platformsSelected: string[];
+  policiesSelected: string[];
+}
 
 interface AgentsTableProps {
-  selectedAgents: string[];
-  onChange: (payload: string[]) => void;
+  agentSelection: AgentsSelection;
+  onChange: (payload: AgentsSelection) => void;
 }
 
-const AgentsTableComponent: React.FC<AgentsTableProps> = ({ selectedAgents, onChange }) => {
-  const [pageIndex, setPageIndex] = useState(0);
-  const [pageSize, setPageSize] = useState(5);
-  const [sortField, setSortField] = useState<keyof Agent>('upgraded_at');
-  const [sortDirection, setSortDirection] = useState<Direction>(Direction.asc);
-  const [selectedItems, setSelectedItems] = useState([]);
-  const tableRef = useRef<EuiBasicTable<Agent>>(null);
-
-  const onTableChange: EuiBasicTableProps<Agent>['onChange'] = useCallback(
-    ({ page = {}, sort = {} }) => {
-      const { index: newPageIndex, size: newPageSize } = page;
-
-      const { field: newSortField, direction: newSortDirection } = sort;
-
-      setPageIndex(newPageIndex);
-      setPageSize(newPageSize);
-      setSortField(newSortField);
-      setSortDirection(newSortDirection);
-    },
-    []
-  );
+type GroupOption = EuiComboBoxOptionOption<AgentOptionValue | GroupOptionValue>;
 
-  const onSelectionChange: EuiTableSelectionType<{}>['onSelectionChange'] = useCallback(
-    (newSelectedItems) => {
-      setSelectedItems(newSelectedItems);
+const getColor = generateColorPicker();
 
-      if (onChange) {
-        // @ts-expect-error update types
-        onChange(newSelectedItems.map((item) => item._id));
-      }
-    },
-    [onChange]
-  );
+const AgentsTableComponent: React.FC<AgentsTableProps> = ({ onChange }) => {
+  // handle paged fetching of agents
+  const [pageIndex /* , setPageIndex*/] = useState(0);
+  const [pageSize /* , setPageSize*/] = useState(1000);
+  const [sortField /* , setSortField*/] = useState<keyof Agent>('upgraded_at');
+  const [sortDirection /* , setSortDirection*/] = useState<Direction>(Direction.asc);
 
-  const renderStatus = (online: string) => {
-    const color = online ? 'success' : 'danger';
-    const label = online ? 'Online' : 'Offline';
-    return <EuiHealth color={color}>{label}</EuiHealth>;
-  };
+  const { loading: groupsLoading, totalCount: totalNumAgents, groups } = useAgentGroups();
+  const [loading, setLoading] = useState<boolean>(true);
+  const [options, setOptions] = useState<GroupOption[]>([]);
+  const [selectedOptions, setSelectedOptions] = useState<GroupOption[]>([]);
+  const [numAgentsSelected, setNumAgentsSelected] = useState<number>(0);
 
-  const { data = {} } = useAllAgents({
+  const { data: agentData = { agents: [] } } = useAllAgents({
```

Comment:
can we set default values inside `useAllAgents`, so we don't need to do it everytime we use that hook?
what about replacing search strategy with fleet API? similar to what we did on the server time I think it will be safer to rely on the fleet API rather just pulling information from the index